In [1]:
import pandas as pd
import numpy as np

In [2]:
import plotly.graph_objects as go

In [3]:
import LabData
from LabData.DataLoaders.Loader import Loader

In [4]:
cgmLoader = Loader.get_loader("CGMLoader")

In [5]:
cgm = cgmLoader.get_data()

In [6]:
cgm_df = cgm.df

In [7]:
gdm_df = pd.read_csv('/home/elming/Downloads/GDM_michal.csv')

In [12]:
gdm_df

clock  RegistrationCode            Name  Round    Due Date profiling  \
0   3/12/2017          859116.0         שי הובר    1.0         NaN         V   
1         NaN           64229.0        נטע פורה    2.0         NaN         V   
2         NaN          637617.0        שרית קול    2.0         NaN         V   
3         NaN          757248.0       אטי תל כץ    2.0         NaN         V   
4         NaN          670055.0        נעמה שחר    3.0  29/12/2017         V   
6         NaN          675329.0      מירב מונטה    3.0         NaN         V   
7         NaN          999443.0      אסיה רביבו    4.0   23/1/2017         V   
8         NaN          292486.0        קרן קידר    4.0    6/2/2018         V   
11        NaN          986640.0      גל גרינברג    5.0   25/2/2018         V   
12        NaN          517738.0        ליאת קול    6.0         NaN         V   
13        NaN          521161.0        חני נמיר    6.0         NaN         V   
15        NaN           30398.0    לימור בר עמי    8.0    2/5/2018         V   
16        NaN          394858.0  אנסטסיה יאביץ'    9.0    7/6/2018         V   
17        NaN          485966.0    דגנית ישראלי   10.0   18/9/1978         V   
18        NaN          471439.0       מור קוקוי   10.0    2/9/2018         V   
19        NaN          909576.0        אדוה שרס   11.0    6/9/2018         V   
20        NaN          405661.0       נעמי הדיה    NaN  20/12/2018         V   
21        NaN          531067.0     ססקיה פרויד    NaN    1/1/2019         V   
22        NaN               NaN         אסנת שץ    NaN         NaN         V   

           T=0   2w   1m   2m  3m labour date 6w al 3m al 6m al  12m al  \
0    10/7/2017    V    V  NaN NaN  2017-08-29     V     V     V     NaN   
1    28/9/2017    V    V  NaN NaN  2017-10-11   NaN     V   NaN   29.11   
2    28/9/2017    V    V  NaN NaN  2017-11-16   NaN     V   NaN     NaN   
3    28/9/2017    V    V    V NaN  2017-11-16   NaN     V     V    3.10   
4   18/10/2017    V    V    V NaN  2017-12-20   NaN     V     V     NaN   
6   18/10/2017    V    V  NaN NaN  2017-03-12   NaN     V     V     NaN   
7   28/11/2017    V    V  NaN NaN  2018-01-13   NaN     V     V     NaN   
8   30/11/2017    V    V    V NaN  2018-04-02   NaN   NaN   NaN     NaN   
11    2/1/2017    V    V  NaN NaN  2018-02-14   NaN     V   NaN     NaN   
12    4/2/2018    V    V  NaN NaN  2018-03-29   NaN     V   NaN     NaN   
13    7/2/2018    V  NaN  NaN NaN  2018-03-25   NaN     V     V     NaN   
15    6/3/2018    V    V  NaN NaN  2018-01-04   NaN     V   NaN     NaN   
16   18/4/2018    V    V  NaN NaN  2018-08-06   NaN   NaN   NaN     NaN   
17    3/7/2018    V    V  NaN NaN  2018-08-24   NaN   3.1   NaN     NaN   
18    1/7/2018  NaN    V  NaN NaN  2018-08-23   NaN     V   NaN     NaN   
19   17/7/2018    V    V  NaN NaN  2018-01-09   NaN   3.1   NaN     NaN   
20  28/10/2018    V    V  NaN NaN  2018-12-22   NaN   NaN   NaN     NaN   
21  13/11/2018    V    V  NaN NaN  2019-10-01   NaN   NaN   NaN     NaN   
22    3/1/2019  NaN  NaN  NaN NaN  2019-02-14   NaN   NaN   NaN     NaN   

   Dropout reason  
0             NaN  
1             NaN  
2             NaN  
3             NaN  
4             NaN  
6             NaN  
7             NaN  
8             NaN  
11            NaN  
12            NaN  
13            NaN  
15            NaN  
16            NaN  
17            NaN  
18            NaN  
19            NaN  
20            NaN  
21            NaN  
22            NaN

In [9]:
gdm_df.dropna(subset=['labour date'], inplace=True)

In [11]:
gdm_df['labour date'] = pd.to_datetime(gdm_df['labour date'])

In [19]:
rc_list = ['859116', '64229', '637617', '757248', '670055', '675329', '999443', '292486', '986640', '517738', '521161', '30398', '394858', '485966', '471439', '909576', '405661', '531067', '511664']

In [20]:
gdm_df['RegistrationCode'] = rc_list

In [23]:
gdm_df

clock RegistrationCode            Name  Round    Due Date profiling  \
0   3/12/2017           859116         שי הובר    1.0         NaN         V   
1         NaN            64229        נטע פורה    2.0         NaN         V   
2         NaN           637617        שרית קול    2.0         NaN         V   
3         NaN           757248       אטי תל כץ    2.0         NaN         V   
4         NaN           670055        נעמה שחר    3.0  29/12/2017         V   
6         NaN           675329      מירב מונטה    3.0         NaN         V   
7         NaN           999443      אסיה רביבו    4.0   23/1/2017         V   
8         NaN           292486        קרן קידר    4.0    6/2/2018         V   
11        NaN           986640      גל גרינברג    5.0   25/2/2018         V   
12        NaN           517738        ליאת קול    6.0         NaN         V   
13        NaN           521161        חני נמיר    6.0         NaN         V   
15        NaN            30398    לימור בר עמי    8.0    2/5/2018         V   
16        NaN           394858  אנסטסיה יאביץ'    9.0    7/6/2018         V   
17        NaN           485966    דגנית ישראלי   10.0   18/9/1978         V   
18        NaN           471439       מור קוקוי   10.0    2/9/2018         V   
19        NaN           909576        אדוה שרס   11.0    6/9/2018         V   
20        NaN           405661       נעמי הדיה    NaN  20/12/2018         V   
21        NaN           531067     ססקיה פרויד    NaN    1/1/2019         V   
22        NaN           511664         אסנת שץ    NaN         NaN         V   

           T=0   2w   1m   2m  3m labour date 6w al 3m al 6m al  12m al  \
0    10/7/2017    V    V  NaN NaN  2017-08-29     V     V     V     NaN   
1    28/9/2017    V    V  NaN NaN  2017-10-11   NaN     V   NaN   29.11   
2    28/9/2017    V    V  NaN NaN  2017-11-16   NaN     V   NaN     NaN   
3    28/9/2017    V    V    V NaN  2017-11-16   NaN     V     V    3.10   
4   18/10/2017    V    V    V NaN  2017-12-20   NaN     V     V     NaN   
6   18/10/2017    V    V  NaN NaN  2017-03-12   NaN     V     V     NaN   
7   28/11/2017    V    V  NaN NaN  2018-01-13   NaN     V     V     NaN   
8   30/11/2017    V    V    V NaN  2018-04-02   NaN   NaN   NaN     NaN   
11    2/1/2017    V    V  NaN NaN  2018-02-14   NaN     V   NaN     NaN   
12    4/2/2018    V    V  NaN NaN  2018-03-29   NaN     V   NaN     NaN   
13    7/2/2018    V  NaN  NaN NaN  2018-03-25   NaN     V     V     NaN   
15    6/3/2018    V    V  NaN NaN  2018-01-04   NaN     V   NaN     NaN   
16   18/4/2018    V    V  NaN NaN  2018-08-06   NaN   NaN   NaN     NaN   
17    3/7/2018    V    V  NaN NaN  2018-08-24   NaN   3.1   NaN     NaN   
18    1/7/2018  NaN    V  NaN NaN  2018-08-23   NaN     V   NaN     NaN   
19   17/7/2018    V    V  NaN NaN  2018-01-09   NaN   3.1   NaN     NaN   
20  28/10/2018    V    V  NaN NaN  2018-12-22   NaN   NaN   NaN     NaN   
21  13/11/2018    V    V  NaN NaN  2019-10-01   NaN   NaN   NaN     NaN   
22    3/1/2019  NaN  NaN  NaN NaN  2019-02-14   NaN   NaN   NaN     NaN   

   Dropout reason  
0             NaN  
1             NaN  
2             NaN  
3             NaN  
4             NaN  
6             NaN  
7             NaN  
8             NaN  
11            NaN  
12            NaN  
13            NaN  
15            NaN  
16            NaN  
17            NaN  
18            NaN  
19            NaN  
20            NaN  
21            NaN  
22            NaN

In [49]:
gdm_df[gdm_df['RegistrationCode'] == '30398']

clock RegistrationCode          Name  Round  Due Date profiling       T=0  \
15   NaN            30398  לימור בר עמי    8.0  2/5/2018         V  6/3/2018   

   2w 1m   2m  3m labour date 6w al 3m al 6m al  12m al Dropout reason  
15  V  V  NaN NaN  2018-01-04   NaN     V   NaN     NaN            NaN

In [24]:
gdm_dict = gdm_df.set_index('RegistrationCode').to_dict()['labour date']

In [25]:
gdm_dict

{'859116': Timestamp('2017-08-29 00:00:00'),
 '64229': Timestamp('2017-10-11 00:00:00'),
 '637617': Timestamp('2017-11-16 00:00:00'),
 '757248': Timestamp('2017-11-16 00:00:00'),
 '670055': Timestamp('2017-12-20 00:00:00'),
 '675329': Timestamp('2017-03-12 00:00:00'),
 '999443': Timestamp('2018-01-13 00:00:00'),
 '292486': Timestamp('2018-04-02 00:00:00'),
 '986640': Timestamp('2018-02-14 00:00:00'),
 '517738': Timestamp('2018-03-29 00:00:00'),
 '521161': Timestamp('2018-03-25 00:00:00'),
 '30398': Timestamp('2018-01-04 00:00:00'),
 '394858': Timestamp('2018-08-06 00:00:00'),
 '485966': Timestamp('2018-08-24 00:00:00'),
 '471439': Timestamp('2018-08-23 00:00:00'),
 '909576': Timestamp('2018-01-09 00:00:00'),
 '405661': Timestamp('2018-12-22 00:00:00'),
 '531067': Timestamp('2019-10-01 00:00:00'),
 '511664': Timestamp('2019-02-14 00:00:00')}

In [27]:
cgm_filtered = cgm_df[cgm_df.index.get_level_values('RegistrationCode').isin(gdm_dict.keys())]

In [108]:
cgm_filtered.index.get_level_values('RegistrationCode').unique()

Index(['292486', '30398', '394858', '405661', '471439', '485966', '511664',
       '517738', '521161', '531067', '637617', '64229', '670055', '675329',
       '757248', '859116', '909576', '986640', '999443'],
      dtype='object', name='RegistrationCode')

In [30]:
mage_df_unfiltered = cgmLoader.calculate_mage(cgm_filtered)

/home/elming/LabRepos/LabData/LabData/DataLoaders/CGMLoader.py:403: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

/home/elming/LabRepos/LabData/LabData/DataLoaders/CGMLoader.py:404: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [31]:
mage_df_unfiltered

RegistrationCode       date   mage  hmage  lmage    ge
0              292486 2017-11-08  19.34  23.00  15.67   4.0
47             292486 2017-11-09  32.67  25.67  39.67   6.0
143            292486 2017-11-10  32.25  26.00  38.50   6.0
239            292486 2017-11-11  17.38  15.00  19.75   6.0
334            292486 2017-11-12  15.44  14.00  16.88   9.0
...               ...        ...    ...    ...    ...   ...
5282           999443 2018-07-15  20.00  19.33  20.67   9.0
5378           999443 2018-07-16  30.34  29.67  31.00  11.0
5474           999443 2018-07-17  16.83  14.86  18.80  12.0
5570           999443 2018-07-18  23.23  23.17  23.29  13.0
5666           999443 2018-07-19  21.62  23.50  19.75   8.0

[1574 rows x 6 columns]

In [93]:
mage_df = mage_df_unfiltered.copy()

import plotly.graph_objects as go

fig = go.Figure(data=go.Heatmap(
                   z=[[19, 32, 50, 1], [20, 1, 60, 80, 30], [30, 60, 1, -10, 20]],
                   x=['Day 0', 'Day 1', 'Day 2', 'Day 3', 'Day4'],
                   y=['rc292486', 'rc999443', 'rc223456'],
                   hoverongaps = False))
fig.show()

In [94]:
mage_df.RegistrationCode.unique()

array(['292486', '30398', '394858', '405661', '471439', '485966',
       '511664', '517738', '521161', '531067', '637617', '64229',
       '670055', '675329', '757248', '859116', '909576', '986640',
       '999443'], dtype=object)

Now I want to have a mage_df with only dates before birth for each participant

I am looking at participant #30398, she doesn't have her CGM data in the CGM_loader before her date of birth. See below

In [95]:
test = mage_df[mage_df['RegistrationCode'] =='30398']
test

RegistrationCode       date   mage  hmage  lmage    ge
0               30398 2018-02-15  10.34   0.00  20.67   3.0
43              30398 2018-02-16  20.25  23.00  17.50   4.0
127             30398 2018-02-17  19.00  19.00  19.00   4.0
213             30398 2018-02-18  10.34  10.00  10.67   6.0
294             30398 2018-02-19  24.84  25.00  24.67   4.0
380             30398 2018-02-20  12.82  12.40  13.25   9.0
472             30398 2018-02-21  10.25   0.00  20.50   4.0
562             30398 2018-02-22  12.00   0.00  24.00   2.0
645             30398 2018-02-23   6.00  12.00   0.00   1.0
691             30398 2018-02-24   0.00   0.00   0.00   0.0
722             30398 2018-02-25  19.00  22.00  16.00   3.0
764             30398 2018-02-26   6.75   0.00  13.50   2.0
817             30398 2018-02-27   6.25   0.00  12.50   2.0
873             30398 2018-02-28   5.00   0.00  10.00   1.0
903             30398 2018-03-07   0.00   0.00   0.00   0.0
939             30398 2018-03-08  16.84  17.00  16.67   4.0
972             30398 2018-03-09   7.50   0.00  15.00   1.0
1042            30398 2018-03-10   4.50   0.00   9.00   1.0
1073            30398 2018-03-12   9.00   0.00  18.00   1.0
1159            30398 2018-03-13  10.00   0.00  20.00   2.0
1220            30398 2018-03-15   9.00   0.00  18.00   1.0
1274            30398 2018-03-16   0.00   0.00   0.00   0.0
1303            30398 2018-03-21   8.00   0.00  16.00   2.0
1365            30398 2018-03-22   8.00   0.00  16.00   3.0
1453            30398 2018-03-23  12.00   0.00  24.00   1.0
1484            30398 2018-03-24   8.50   0.00  17.00   1.0
1532            30398 2018-03-25   9.00   0.00  18.00   3.0
1593            30398 2018-03-26  27.25  26.00  28.50   3.0
1659            30398 2018-03-27  21.50  25.00  18.00   3.0
1744            30398 2018-03-28  17.25  17.00  17.50   3.0
1830            30398 2018-03-29  13.00   0.00  26.00   1.0
1861            30398 2018-03-30  18.00  15.00  21.00   3.0
1892            30398 2018-03-31   8.50   0.00  17.00   1.0
1957            30398 2018-04-01   4.50   9.00   0.00   3.0
1987            30398 2018-04-22  15.50  15.00  16.00  11.0
2037            30398 2018-04-23  25.75  26.50  25.00  10.0
2129            30398 2018-04-24  30.48  30.83  30.14  13.0
2221            30398 2018-04-25  29.12  28.25  30.00   7.0
2316            30398 2018-04-26  22.50  22.00  23.00   9.0
2409            30398 2018-04-27  25.25  23.00  27.50   4.0
2502            30398 2018-04-28  12.00  24.00   0.00   1.0
2597            30398 2018-04-29  22.50  21.00  24.00   3.0
2693            30398 2018-04-30  26.50  28.00  25.00   2.0
2789            30398 2018-05-01   5.00   5.00   5.00   3.0

In [96]:
gdm_df[gdm_df['RegistrationCode'] == '30398']['labour date']

15   2018-01-04
Name: labour date, dtype: datetime64[ns]

Here you see that in our loader we don't find the CGM data before she gave the birth on 2018-01-04

In [97]:
cgm_df.loc[('30398', slice(None), slice(None)), :]

GlucoseValue  PPGR
RegistrationCode ConnectionID Date                                         
30398            2466         2018-02-15 10:50:00+02:00          77.0   NaN
                              2018-02-15 11:05:00+02:00          75.0   NaN
                              2018-02-15 11:20:00+02:00          67.0   NaN
                              2018-02-15 11:35:00+02:00          61.0   NaN
                              2018-02-15 11:50:00+02:00          57.0   NaN
...                                                               ...   ...
                 5543         2018-05-01 08:12:00+03:00          83.0   NaN
                              2018-05-01 08:27:00+03:00          81.0   NaN
                              2018-05-01 08:42:00+03:00          80.0   NaN
                              2018-05-01 08:57:00+03:00          80.0   NaN
                              2018-05-01 09:12:00+03:00          82.0   NaN

[2826 rows x 2 columns]

So it makes sense now that the function below drops all the people that don't have CGM data before giving a birth

In [98]:
for rc in mage_df["RegistrationCode"].unique():
    
    indexNames = mage_df[ (mage_df['RegistrationCode'] == rc) & (mage_df['date'] >= gdm_dict[rc]) ].index
    mage_df.drop(indexNames, inplace=True)
mage_df['RegistrationCode'].unique()

array(['292486', '394858', '405661', '471439', '485966', '511664',
       '517738', '521161', '531067', '637617', '64229', '670055',
       '757248', '859116', '986640', '999443'], dtype=object)

In [99]:
len(mage_df['RegistrationCode'].unique())

16

Now I will proceed with a heatmap for these 16 women. First, I need to find out maximum days of connection a person had

In [100]:
for rc in mage_df["RegistrationCode"].unique():
    print(rc, len(mage_df[mage_df['RegistrationCode'] == rc]))

292486 48
394858 54
405661 32
471439 53
485966 53
511664 57
517738 34
521161 52
531067 64
637617 48
64229 26
670055 46
757248 54
859116 39
986640 46
999443 47


Ok, the maximum number of connection days befor labour date (at least in the data that is inside the cgm loader) was 64 for the participant 531067. Now I will make the list of day names for the heatmap.

In [101]:
list_of_days = []
for i in range(64):
    list_of_days.append('Day {}'.format(i+1))
print(list_of_days)

['Day 1', 'Day 2', 'Day 3', 'Day 4', 'Day 5', 'Day 6', 'Day 7', 'Day 8', 'Day 9', 'Day 10', 'Day 11', 'Day 12', 'Day 13', 'Day 14', 'Day 15', 'Day 16', 'Day 17', 'Day 18', 'Day 19', 'Day 20', 'Day 21', 'Day 22', 'Day 23', 'Day 24', 'Day 25', 'Day 26', 'Day 27', 'Day 28', 'Day 29', 'Day 30', 'Day 31', 'Day 32', 'Day 33', 'Day 34', 'Day 35', 'Day 36', 'Day 37', 'Day 38', 'Day 39', 'Day 40', 'Day 41', 'Day 42', 'Day 43', 'Day 44', 'Day 45', 'Day 46', 'Day 47', 'Day 48', 'Day 49', 'Day 50', 'Day 51', 'Day 52', 'Day 53', 'Day 54', 'Day 55', 'Day 56', 'Day 57', 'Day 58', 'Day 59', 'Day 60', 'Day 61', 'Day 62', 'Day 63', 'Day 64']


Now I want to make a list of lists with mage values for each person per day. If the number of days for a person was smaller than 64, then it should be filled with NaN(None).

In [102]:
list_of_mage_values = []
for rc in mage_df["RegistrationCode"].unique():
    list_of_mage_values.append(list(mage_df[mage_df['RegistrationCode'] == rc]['mage']))

In [103]:
print(list_of_mage_values)

[[32.67, 32.25, 15.44, 22.0, 21.91, 14.75, 20.12, 25.2, 16.8, 0.0, 28.75, 19.36, 18.83, 19.1, 14.0, 7.12, 9.25, 13.25, 9.12, 15.16, 8.75, 7.0, 8.84, 6.5, 16.75, 19.34, 9.16, 12.96, 8.34, 10.84, 8.0, 7.5, 6.84, 8.8, 20.84, 41.0, 18.75, 16.5, 9.0, 13.4, 12.5, 16.25, 14.66, 14.5, 7.0, 15.12, 13.5, 6.0], [26.25, 14.0, 15.8, 18.75, 21.75, 16.8, 9.84, 8.8, 8.34, 13.12, 24.0, 6.12, 14.38, 9.5, 18.59, 9.62, 52.0, 17.88, 13.5, 17.84, 10.0, 14.25, 14.75, 19.5, 5.5, 12.0, 13.71, 18.5, 0.0, 16.12, 10.75, 10.25, 9.5, 12.66, 0.0, 0.0, 18.09, 20.86, 25.5, 19.0, 15.5, 17.25, 19.64, 10.4, 12.0, 13.25, 5.5, 8.5, 8.34, 18.0, 16.25, 9.5, 17.0, 2.5], [18.34, 16.7, 14.0, 15.84, 19.16, 17.29, 8.16, 20.5, 13.0, 27.5, 21.25, 16.96, 19.0, 18.9, 8.5, 19.88, 11.34, 9.5, 9.5, 15.84, 7.25, 8.12, 13.38, 6.0, 9.75, 11.84, 17.0, 14.62, 17.0, 11.84, 6.5, 8.75], [25.0, 12.25, 28.16, 12.25, 20.75, 21.25, 16.25, 17.09, 16.0, 17.75, 10.5, 14.0, 6.5, 13.75, 11.8, 14.16, 19.25, 19.5, 17.15, 28.84, 13.0, 33.25, 16.5, 12.16, 1

In [104]:
reg_code_list = ['rc'+i for i in mage_df['RegistrationCode'].unique()]

In [105]:
reg_code_list.sort()

In [106]:
reg_code_list

['rc292486',
 'rc394858',
 'rc405661',
 'rc471439',
 'rc485966',
 'rc511664',
 'rc517738',
 'rc521161',
 'rc531067',
 'rc637617',
 'rc64229',
 'rc670055',
 'rc757248',
 'rc859116',
 'rc986640',
 'rc999443']

In [107]:
fig = go.Figure(data=go.Heatmap(
                   z=list_of_mage_values,
                   x=list_of_days,
                   y=reg_code_list,
                   hoverongaps = False))
fig.update_layout(
    title='MAGE before labour date')
fig.show()

The problem here is that we don't know which period of pregnancy corresponds to which day, we only kno that all the days shown here are before giving a birth. Maybe let's look on the heatmap also at the days after the labour day

In [74]:
mage_df = mage_df_unfiltered.copy()

In [75]:
for rc in mage_df["RegistrationCode"].unique():
    
    indexNames = mage_df[ (mage_df['RegistrationCode'] == rc) & (mage_df['date'] <= gdm_dict[rc]) ].index
    mage_df.drop(indexNames, inplace=True)
mage_df['RegistrationCode'].unique()

array(['30398', '394858', '405661', '471439', '485966', '511664',
       '521161', '637617', '64229', '670055', '675329', '757248',
       '859116', '909576', '986640', '999443'], dtype=object)

In [76]:
len(mage_df['RegistrationCode'].unique())

16

In [77]:
for rc in mage_df["RegistrationCode"].unique():
    print(rc, len(mage_df[mage_df['RegistrationCode'] == rc]))

30398 37
394858 42
405661 21
471439 28
485966 147
511664 14
521161 36
637617 13
64229 64
670055 24
675329 31
757248 58
859116 68
909576 47
986640 32
999443 23


The max is 147, but let's only take 40 days for each person

In [85]:
list_of_days = []
for i in range(40):
    list_of_days.append('Day {}'.format(i+1))
print(list_of_days)

['Day 1', 'Day 2', 'Day 3', 'Day 4', 'Day 5', 'Day 6', 'Day 7', 'Day 8', 'Day 9', 'Day 10', 'Day 11', 'Day 12', 'Day 13', 'Day 14', 'Day 15', 'Day 16', 'Day 17', 'Day 18', 'Day 19', 'Day 20', 'Day 21', 'Day 22', 'Day 23', 'Day 24', 'Day 25', 'Day 26', 'Day 27', 'Day 28', 'Day 29', 'Day 30', 'Day 31', 'Day 32', 'Day 33', 'Day 34', 'Day 35', 'Day 36', 'Day 37', 'Day 38', 'Day 39', 'Day 40']


Now I want to make a list of lists with mage values for each person per day. If the number of days for a person was smaller than 64, then it should be filled with NaN(None).

In [86]:
list_of_mage_values = []
for rc in mage_df["RegistrationCode"].unique():
    list_of_mage_values.append(list(mage_df[mage_df['RegistrationCode'] == rc]['mage'])[0:41])

In [87]:
print(list_of_mage_values)

[[20.25, 19.0, 10.34, 24.84, 12.82, 10.25, 12.0, 6.0, 0.0, 19.0, 6.75, 6.25, 5.0, 0.0, 16.84, 4.5, 9.0, 0.0, 8.0, 8.0, 12.0, 8.5, 9.0, 27.25, 17.25, 13.0, 18.0, 8.5, 4.5, 15.5, 30.48, 29.12, 22.5, 25.25, 12.0, 26.5, 5.0], [15.0, 30.25, 16.5, 70.5, 21.5, 16.89, 18.5, 25.25, 28.84, 17.16, 12.25, 23.0, 19.84, 3.25, 30.0, 28.25, 18.0, 13.25, 14.34, 9.46, 25.0, 15.0, 27.75, 42.0, 0.0, 20.02, 15.0, 17.5, 17.5, 15.5, 11.5, 24.34, 17.79, 15.41, 16.96, 12.75, 25.5, 21.75, 30.59, 24.5, 9.4], [26.5, 25.75, 17.41, 27.75, 23.18, 15.41, 27.5, 22.62, 21.0, 22.0, 16.0, 35.75, 3.0, 14.62, 17.11, 13.5, 22.04, 18.75, 15.5, 26.5, 0.0], [16.25, 25.12, 12.5, 35.5, 21.5, 5.25, 20.0, 0.0, 11.66, 15.0, 16.5, 13.5, 14.5, 6.0, 14.5, 10.5, 22.5, 24.66, 17.95, 14.5, 24.25, 8.5, 12.75, 21.91, 11.0, 13.84, 15.25, 2.0], [9.25, 21.75, 32.75, 25.0, 14.69, 23.0, 17.5, 21.66, 19.85, 11.66, 18.75, 28.84, 11.0, 22.0, 22.6, 42.75, 11.0, 20.0, 20.0, 16.5, 18.88, 29.5, 18.0, 16.25, 26.88, 0.0, 16.16, 18.88, 19.84, 9.0, 11.0, 

In [88]:
reg_code_list = ['rc'+i for i in mage_df['RegistrationCode'].unique()]

In [89]:
reg_code_list.sort()

In [90]:
reg_code_list

['rc30398',
 'rc394858',
 'rc405661',
 'rc471439',
 'rc485966',
 'rc511664',
 'rc521161',
 'rc637617',
 'rc64229',
 'rc670055',
 'rc675329',
 'rc757248',
 'rc859116',
 'rc909576',
 'rc986640',
 'rc999443']

In [92]:
fig = go.Figure(data=go.Heatmap(
                   z=list_of_mage_values,
                   x=list_of_days,
                   y=reg_code_list,
                   hoverongaps = False))
fig.update_layout(
    title='MAGE after labour date')
fig.show()